packages/core/src/lib/__tests__/code-snippet.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { describe, it, expect } from 'vitest';
+import { render, screen } from '@testing-library/svelte';
+import { CodeSnippet } from '$lib';
+import { cxTestArguments, cxTestResults } from './cx-test';
+import CaptionedCodeSnippet from './code-snippet.spec.svelte';
+
+describe('CodeSnippet', () => {
+  const common = {
+    language: 'json',
+    code: '{ my: "json" }',
+  };
+
+  it('Renders the code snippet', () => {
+    render(CodeSnippet, common);
+    expect(screen.getByText('{ my: "json" }')).toBeInTheDocument();
+  });
+
+  it('Renders with the passed cx classes', () => {
+    render(CodeSnippet, {
+      ...common,
+      cx: cxTestArguments,
+    });
+
+    expect(screen.getByRole('figure')).toHaveClass(cxTestResults);
+  });
+
+  it('Renders with the copy button', () => {
+    render(CodeSnippet, common);
+
+    expect(screen.getByLabelText('Copy')).toBeInTheDocument();
```

Comment:
Can this:

- Check for a button with a name rather than the label
- Actually test the copy functionality?

Reply:
On the first point sure, on the second point I guess but IIRC doesn't jsdom not really support the `navigator.clipboard` API. I feel like I recall having to put in some hacky workarounds/mocks, but not sure if something has changed there. I can put something in there if we want though.